File: csharp/QuickTrace/Measurement.cs
namespace QuickTrace;

/// <summary>
/// Represents a single measurement with statement and duration
/// </summary>
public readonly struct Measurement
{
    public string Statement { get; }
    public TimeSpan Duration { get; }

    public Measurement(string statement, TimeSpan duration)
    {
        Statement = statement;
        Duration = duration;
    }

    public override string ToString()
    {
        return $"{Statement}: {Duration}";
    }
}

File: java/src/main/java/com/leduy/quicktrace/Measurement.java
package com.leduy.quicktrace;

import java.time.Duration;

/**
 * Đại diện cho một measurement trong trace
 */
public class Measurement {
    private final String statement;
    private final Duration duration;
    
    public Measurement(String statement, Duration duration) {
        this.statement = statement;
        this.duration = duration;
    }
    
    public String getStatement() {
        return statement;
    }
    
    public Duration getDuration() {
        return duration;
    }
    
    @Override
    public String toString() {
        return String.format("Measurement{statement='%s', duration=%s}", statement, duration);
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        
        Measurement that = (Measurement) o;
        return statement.equals(that.statement) && duration.equals(that.duration);
    }
    
    @Override
    public int hashCode() {
        return statement.hashCode() * 31 + duration.hashCode();
    }
}
